# DATA2001 Assingment 1

## Objectives 
1. Identify Missing Data and Handle
2. Visualise Data
3. Investigate Differences between Species 
4. Forcast using k-Means and PCA

## Table of Contents

Todo

# 1. Problem Formation

# 2. Data Collection

Data were collected and made available by Dr. Kristen Gorman and the Palmer Station, Antarctica LTER, a member of the Long Term Ecological Research Network.There are 3 different species of penguins in this dataset, collected from 3 islands in the Palmer Archipelago, Antarctica.


1. **studyName**- Sampling expedition from which data were collected, generated, etc.

2. **Sample Number** - an integer denoting the continuous numbering sequence for each sample

3. **Species**- a character string denoting the penguin species

4. **Region** - a character string denoting the region of Palmer LTER sampling grid

5. **Island** - a character string denoting the island near Palmer Station where samples were collected

6. **Stage** - a character string denoting reproductive stage at sampling

7. **Individual ID** - a character string denoting the unique ID for each individual in dataset

8. **Clutch Completion** - a character string denoting if the study nest observed with a full clutch, i.e., 2 eggs

9. **Date Egg**- a date denoting the date study nest observed with 1 egg (sampled)

10. **Culmen Length**- a number denoting the length of the dorsal ridge of a bird's bill (millimetres)

11. **Culmen Depth** - a number denoting the depth of the dorsal ridge of a bird's bill (millimetres)

12. **Flipper Length**- an integer denoting the length penguin flipper (millimetres)

13. **Body Mass**- an integer denoting the penguin body mass (grams)

14. **Sex** - a character string denoting the sex of an animal

15. **Delta 15 N**- a number denoting the measure of the ratio of stable isotopes 15N:14N

16. **Delta 13 C** - a number denoting the measure of the ratio of stable isotopes 13C:12C

17. **Comments** - a character string with text providing additional relevant information for data

## 2.1 Load penguins_iter.csv



In [103]:
# import pandas
import pandas as pd

#Supresss warnings for neatness
import warnings
warnings.filterwarnings("ignore")

#load data
df = pd.read_csv('penguins_lter.csv')

#view head
df.head(5)

studyName  Sample Number                              Species  Region  \
0   PAL0708              1  Adelie Penguin (Pygoscelis adeliae)  Anvers   
1   PAL0708              2  Adelie Penguin (Pygoscelis adeliae)  Anvers   
2   PAL0708              3  Adelie Penguin (Pygoscelis adeliae)  Anvers   
3   PAL0708              4  Adelie Penguin (Pygoscelis adeliae)  Anvers   
4   PAL0708              5  Adelie Penguin (Pygoscelis adeliae)  Anvers   

      Island               Stage Individual ID Clutch Completion  Date Egg  \
0  Torgersen  Adult, 1 Egg Stage          N1A1               Yes  11/11/07   
1  Torgersen  Adult, 1 Egg Stage          N1A2               Yes  11/11/07   
2  Torgersen  Adult, 1 Egg Stage          N2A1               Yes  11/16/07   
3  Torgersen  Adult, 1 Egg Stage          N2A2               Yes  11/16/07   
4  Torgersen  Adult, 1 Egg Stage          N3A1               Yes  11/16/07   

   Culmen Length (mm)  Culmen Depth (mm)  Flipper Length (mm)  Body Mass (g)  \
0                39.1               18.7                181.0         3750.0   
1                39.5               17.4                186.0         3800.0   
2                40.3               18.0                195.0         3250.0   
3                 NaN                NaN                  NaN            NaN   
4                36.7               19.3                193.0         3450.0   

      Sex  Delta 15 N (o/oo)  Delta 13 C (o/oo)  \
0    MALE                NaN                NaN   
1  FEMALE            8.94956          -24.69454   
2  FEMALE            8.36821          -25.33302   
3     NaN                NaN                NaN   
4  FEMALE            8.76651          -25.32426   

                         Comments  
0  Not enough blood for isotopes.  
1                             NaN  
2                             NaN  
3              Adult not sampled.  
4                             NaN

### Missing Values

In [104]:
df.describe()

Sample Number  Culmen Length (mm)  Culmen Depth (mm)  \
count     344.000000          342.000000         342.000000   
mean       63.151163           43.921930          17.151170   
std        40.430199            5.459584           1.974793   
min         1.000000           32.100000          13.100000   
25%        29.000000           39.225000          15.600000   
50%        58.000000           44.450000          17.300000   
75%        95.250000           48.500000          18.700000   
max       152.000000           59.600000          21.500000   

       Flipper Length (mm)  Body Mass (g)  Delta 15 N (o/oo)  \
count           342.000000     342.000000         330.000000   
mean            200.915205    4201.754386           8.733382   
std              14.061714     801.954536           0.551770   
min             172.000000    2700.000000           7.632200   
25%             190.000000    3550.000000           8.299890   
50%             197.000000    4050.000000           8.652405   
75%             213.000000    4750.000000           9.172123   
max             231.000000    6300.000000          10.025440   

       Delta 13 C (o/oo)  
count         331.000000  
mean          -25.686292  
std             0.793961  
min           -27.018540  
25%           -26.320305  
50%           -25.833520  
75%           -25.062050  
max           -23.787670

In [105]:
print("Percentage of missing values by column:")
#Percentage missing values by column
df.isnull().sum()/len(df)*100

Percentage of missing values by column:


studyName               0.000000
Sample Number           0.000000
Species                 0.000000
Region                  0.000000
Island                  0.000000
Stage                   0.000000
Individual ID           0.000000
Clutch Completion       0.000000
Date Egg                0.000000
Culmen Length (mm)      0.581395
Culmen Depth (mm)       0.581395
Flipper Length (mm)     0.581395
Body Mass (g)           0.581395
Sex                     2.906977
Delta 15 N (o/oo)       4.069767
Delta 13 C (o/oo)       3.779070
Comments               92.441860
dtype: float64

In [106]:
rows = df.shape[0]
__ = df.drop('Comments', axis=1)
print('Rows missing values (excluding comments):', rows-__.dropna().shape[0])

Rows missing values (excluding comments): 19


In [108]:
#save comment row 
comments = df['Comments']
df = df.drop('Comments', axis=1)

#replace all other missing values with the mean of the column 
df = df.fillna(df.median())
df['Comments'] = comments
df = df.fillna('No comment')

### Missing values and imputation 

92% of the rows are missing comments. I am going to assume that we can keep these rows and replace the missing comments with a place holder as comments appear to be optional. There are 19 rows that are missing values exluding comments. These are all continuous data types. Replacing these with the median appears to be appropriate given the nature of the distribution of the missing values. In this case imputing with the median as opposed to the mean was done in hopes that the use of the median mitigates the effect of outliers on the analysis. However, median or mean would probably both be suitable in the context of k-means and PCA. 

# 3. Exploratory Data Analysis (EDA)

Including univariate and bivariate EDA for all relevant columns.  

